Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2771,6 +2771,41 @@ bool CSSPropertyParser::consumeWhiteSpaceShorthand(bool important)
     return true;
 }
 
+static RefPtr<CSSValue> rangeEndValueForCSSValueID(CSSValueID valueID)
```

Comment:
I think this method could be more clear (and potentially just a lambda), `rangeEndValueForStartValue` could be a good name.

If there's a way to write this more similarly to the spec:
> If [<'animation-range-end'>](https://drafts.csswg.org/scroll-animations-1/#propdef-animation-range-end) is omitted and [<'animation-range-start'>](https://drafts.csswg.org/scroll-animations-1/#propdef-animation-range-start) includes a [<timeline-range-name>](https://drafts.csswg.org/scroll-animations-1/#typedef-timeline-range-name) component, then animation-range-end is set to that same <timeline-range-name> and 100%. Otherwise, any omitted [longhand](https://drafts.csswg.org/css-cascade-5/#longhand) is set to its [initial value](https://drafts.csswg.org/css-cascade-5/#initial-value).

it would be good

Reply:
Fixed.